### ISSUE TITLE
Rcon not working with anything?

### ISSUE THREAD

**Issue by mooseh** [xPaw/PHP-Source-Query#109]
Hi There, ive been testing php source query and i seem to not be able to send ANY rcon commands to my servers, being rust/csgo/insurgency below is my code, ping works getinfo work players work but rcon just doesnt work, I used to use this with no issues?

has something changed to make this not work anymore?

I tested the udp connection from my webserver to and it has no problems what so ever. I also tested terminal RCON with no issues im using php7.0

`    public function Rcon(Server $server, Request $request)
    {
        // For the sake of this example
	Header( 'Content-Type: text/plain' );
	Header( 'X-Content-Type-Options: nosniff' );

        $rconPass = decrypt($server->rcon);

        $command = $request->command;

        $Query = new SourceQuery();

        define( 'SQ_SERVER_ADDR', $server->ip );
        define( 'SQ_SERVER_PORT', $server->port );
        define( 'SQ_TIMEOUT',     1 );
        define( 'SQ_ENGINE',      SourceQuery::SOURCE );        

        try
            {
                $Query->Connect( SQ_SERVER_ADDR, SQ_SERVER_PORT, SQ_TIMEOUT, SQ_ENGINE );
                
                $Query->SetRconPassword( $rconPass );
                
                var_dump( $Query->Rcon( 'say hello' ) );
            }
            catch( Exception $e )
            {
                echo $e->getMessage( );
            }
            finally
            {
                $Query->Disconnect( );
            }`

stack trace for the issue is

InvalidPacketException in SourceRcon.php line 100:
Rcon read: Failed to read any data from socket

in SourceRcon.php line 100
at SourceRcon->Read() in SourceRcon.php line 197
at SourceRcon->Authorize('myrconpass') in SourceQuery.php line 553

**Comment by mooseh** [comment]
any response on this? I really do love your SourceQuery, unfortunately I just dont understand fsocket enough to fix this, I tested your old code from the same machine and it works, i would prefer not to work with the old version as i would have to remap and namespace it all.

Any help given is much appreciated.

Thanks

**Comment by xPaw** [comment]
It's very possible that rcon broke during the rewrite I was doing. Unfortunately I currently don't have any time to look into it and fix it.

**Comment by pkminer30** [comment]
SourceQuery/SourceRcon.php line 140 should be commented to work with rust :)

**Comment by Gnako** [comment]
I've been struggling with this too, and found out by parsing the RCONPASSWORD as a variable creates the issue for some reason.
When I tried to execute with the PW directly as a string => "PASSWORD" instead of using the variable it worked fine. Even tried to force the PW to be sent as a String but no difference...

Am I missing something or what could be causing this issue?
INFO: Yes I did check so  the pw that is sent to setRconPassword is correct by echo it out..
